### PR TITLE
test: add tests for Modal

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
Adds tests for Modal

**Pending:** 
- [ ] Add tests for native

### CODE REVIEW

File: packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+          </Button>
+        </>
+      );
+    };
+
+    const { getByRole, queryByRole } = renderWithTheme(<Element />);
+    await user.click(getByRole('button'));
+    expect(queryByRole('dialog')).toBeInTheDocument();
+    await user.keyboard('{Escape}');
+    expect(onDismiss).toHaveBeenCalledTimes(1);
```
Comment (divyanshu013) [inline packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx:129]: You may also expect the dialog to not be there anymore after this point

File: packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            Toggle Modal
+          </Button>
+        </>
+      );
+    };
+
+    const { getByRole, getByTestId } = renderWithTheme(<Element />);
+    await user.click(getByRole('button'));
+    await user.click(getByTestId('modal-backdrop'));
+    expect(onDismiss).toHaveBeenCalledTimes(1);
```
Comment (divyanshu013) [inline packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx:159]: We can also expect the modal to not be rendering anymore

**Review by divyanshu013 — COMMENTED** [review]
Looks good overall, I think similar tests for native would do

File: packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx (hunk 1 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx:128]: There is some issue with the state not changing on jest randomly which made the test very flaky. Initially I was asserting that only but switched to asserting call of `onDismiss`. If I run the test as `it.only` it passes but if I run all of them together it fails when trying to assert dialog's dismount :face

File: packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx (hunk 2 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx:158]: Same as this: https://github.com/razorpay/blade/pull/1251#discussion_r1218955256

File: packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx (hunk 1 above)
Comment (divyanshu013) [reply packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx:128]: Strange, might be due to animation but non-blocker

File: packages/blade/src/components/Modal/__tests__/Modal.native.test.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import React from 'react';
+import { Modal, ModalBody } from '../Modal';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';
+import { Text } from '~components/Typography';
+
+describe('Modal', () => {
+  beforeEach(() => {
+    jest.spyOn(console, 'warn').mockImplementation(() => {});
+  });
```
Comment (divyanshu013) [inline packages/blade/src/components/Modal/__tests__/Modal.native.test.tsx:10]: Let's overwrite this only on appropriate blocks ref #1146

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

**Review by anuraghazra — APPROVED** [review]
Looks good.

**Comment by anuraghazra** [comment]
Tracking on separate issue for useTrailingRestriction tests: https://github.com/razorpay/blade/issues/1270

**Pull request merged**